src/cargo/ops/cargo_add/mod.rs

```diff
@@ -364,7 +364,24 @@ fn resolve_dependency(
     };
     selected_dep = populate_dependency(selected_dep, arg);
 
-    let old_dep = get_existing_dependency(manifest, selected_dep.toml_key(), section)?;
+    let mut old_dep = get_existing_dependency(manifest, selected_dep.toml_key(), section)?;
+    if old_dep.is_none() && selected_dep.source().is_none() && selected_dep.rename().is_none() {
+        for name_permutation in [
+            selected_dep.name.replace('-', "_"),
+            selected_dep.name.replace('_', "-"),
+        ] {
```

Comment:
Should we break this out into a function to be reused between this and `find_workspace_dep`?

Reply:
I did this and pushed a new version with that